### PR TITLE
Adding negative test for plugins with rest endpoints

### PR DESCRIPTION
**Pull request by CuriousVini**
Adding negative test to support change made in https://github.com/caskdata/cdap/pull/8209

build: http://builds.cask.co/browse/CDAP-DUT5536-3

### CODE REVIEW

**Comment by CuriousVini** [comment]
Build is green http://builds.cask.co/browse/CDAP-DUT5536-1

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ArtifactHttpHandlerTest.java [hunk 1]
```diff
@@ -599,6 +599,28 @@ public void testPluginWithEndpoints() throws Exception {
...
+
+    Set<ArtifactRange> plugins5Parents = Sets.newHashSet(new ArtifactRange(
+      NamespaceId.DEFAULT, "wordcount", new ArtifactVersion("1.0.0"), new ArtifactVersion("2.0.0")));
+    Assert.assertEquals(HttpResponseStatus.OK.getCode(),
+                        addPluginArtifact(pluginsId, PluginWithPojo.class, manifest,
+                                          plugins5Parents).getStatusLine().getStatusCode());
+
+    // this call should throw IllegalArgumentException
+    callPluginMethod(pluginsId, "interactive", "aggregator", "throwException", "testString",
+                     ArtifactScope.USER, 400).getResponseBodyAsString();
```
Comment (albertshau) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ArtifactHttpHandlerTest.java]: should you check that the response body is the exception message?

Comment (CuriousVini) [reply cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ArtifactHttpHandlerTest.java]: Yes. Thats right. Fixed it

**Comment by CuriousVini** [comment]
@albertshau Thanks for the review. Updated the test. Could you please take another pass when you get a chance?

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**